pinot-query-planner/src/main/java/org/apache/pinot/query/planner/StageMetadata.java

```diff
@@ -75,43 +73,21 @@ public void addScannedTable(String tableName) {
   // attached physical plan context.
   // -----------------------------------------------
 
-  public Map<ServerInstance, Map<String, List<String>>> getServerInstanceToSegmentsMap() {
-    return _serverInstanceToSegmentsMap;
-  }
-
-  public void setServerInstanceToSegmentsMap(
```

Comment:
all setters should be moved to `DispatchablePlanContext` which gradually completes the stage metadata. 

removing all setter makes StageMetadata immutable